tower/src/util/boxed/sync.rs

```diff
@@ -39,6 +40,15 @@ impl<T, U, E> BoxService<T, U, E> {
         let inner = Box::new(Boxed { inner });
         BoxService { inner }
     }
+
+    #[allow(missing_docs)]
+    pub fn layer<S>() -> impl Layer<S, Service = Self> + Clone
```

Comment:
Since `Self::new` is not a closure, this actually can return a named type rather than an `impl Layer`. I think this could just be
```suggestion
    pub fn layer<S>() -> LayerFn<fn(S) -> Self>
```